components/server/src/user/stripe-service.ts

```diff
@@ -113,6 +113,18 @@ export class StripeService {
             return this.getStripe().subscriptions.del(subscriptionId, { invoice_now: true });
         });
     }
+
+    public async updateAttributionId(stripeCustomerId: string, attributionId: string): Promise<void> {
+        try {
+            await this.getStripe().customers.update(stripeCustomerId, {
+                metadata: {
+                    attributionId,
+                },
+            });
+        } catch (error) {
+            log.error("Error updating Stripe customer metadata", error, { stripeCustomerId, attributionId });
```

Comment:
Why are we swallowing the errors at this level? Shouldn't it be the caller to choose to handle that?

Reply:
yes, would make more sense. This method is not supposed to be used elsewhere, though, but I will move the error handling to the callsite in the follow-up to this (fixing already migrated users) if there are no other reasons to do any other round on this changeset.

Reply:
Then I'd have expected the modifier to be private, but I guess you'd not be able to override it in tests then